### PR TITLE
Minimum changes to support React 16

### PR DESCRIPTION
**Pull request by paularmstrong**
**Problem:** #76 makes too many changes, does much more than just supports React 16.

**Solution:** Apply only the minimum changeset to allow react-globalize to work with React 15.6.x and 16.x.x

**Outstanding issues:** The build system looks like it will actually need to be changed, but separately, because esperanto chokes on babel-plugin-transform-class-properties.

### CODE REVIEW

**Comment by diligiant** [comment]
@paularmstrong aren’t the tests 16.x only? (different enzyme adapter)

**Comment by paularmstrong** [comment]
@diligiant Yes, but I don't see the necessity of creating separate enzyme adapters for 15.6 and 16.x unless there's some sort of regression found. We're using this on Twitter Lite successfully on 15.6 right now.

**Comment by diligiant** [comment]
Then shouldn’t this be R16.x only ? How can “it” claim to be compatible with 15.6 if it is not able to test itself with this version ? you and I know it works because we’ve done the manual changes to be sure but that’s irrelevant to people out there who can’t reproduce easily what we’ve done.

On another topic, what do you have in mind to move forward ? Another PR for Rollup and another one for what’s left ? (I’m new here so I ask ;)

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules/
 dist/
+.DS_Store
```
Comment (necolas) [inline .gitignore]: Remove? People should have this globally ignored

File: src/generator.js [hunk 2]
```diff
@@ -23,15 +22,18 @@ function generator(fn, localPropNames, options) {
     };
     var globalizePropNames = commonPropNames.concat(localPropNames);
 
-    return reactCreateClass({
-        displayName: Fn,
-        componentWillMount: function() {
+    return class extends React.Component {
+        // static displayName = Fn;
```
Comment (necolas) [inline src/generator.js]: Uncomment?

Comment (paularmstrong) [reply src/generator.js]: This is the *Outstanding issues* from the description. Esperanto chokes when this is hit, even with the babel plugin.

Comment (necolas) [reply src/generator.js]: Ah, I see

Comment (rxaviers) [reply src/generator.js]: Does that mean we need the tool improvements too?

**Comment by rxaviers** [comment]
Waiting for @paularmstrong's response on https://github.com/globalizejs/react-globalize/pull/78#issuecomment-334587747

File: src/generator.js (hunk 2 above)
Comment (paularmstrong) [reply src/generator.js]: They will be needed if you want to support `displayName`. But that's really only for debugging purposes, so it's not a blocker for things to work.

**Comment by rxaviers** [comment]
> to work with React 15.6.x and 16.x.x

How should we test 15.x? I believe this is related to #77

**Comment by paularmstrong** [comment]
Apparently grunt/esperanto doesn't actually run babel across the files? That explains the `static` issue noted inline. Also fails for us building production builds, because the class isn't compiled JS.

The build system needs to be overhauled first. ¯\_(ツ)_/¯

**Pull request closed without merging**